### PR TITLE
Initial PR

### PR DESCRIPTION
**Pull request by svenwltr**
> Inspired by https://www.hashicorp.com/blog/managing-google-calendar-with-terraform

**Checklist**

* [x] single initial resource
* [x] acceptance tests
* [x] automatic testing
* [x] README / Documentation

**Purpose**

IMO this should replace https://github.com/rebuy-de/graylog-configurator. Terraform is more convenient for setting stuff up. Also we could replace our other configuration glue scripts with Terraform (see [here](https://www.terraform.io/docs/providers/mysql/index.html), [here](https://www.terraform.io/docs/providers/newrelic/index.html), [here](https://www.terraform.io/docs/providers/postgresql/index.html), [here](https://www.terraform.io/docs/providers/rabbitmq/index.html) and [here](https://www.terraform.io/docs/providers/grafana/index.html)). The only thing missing is a neat way for automatic deployments in our cluster.

### CODE REVIEW

**Comment by svenwltr** [comment]
@rebuy-de/it-platform What do you think?

**Comment by stephanlindauer** [comment]
looks good. another candidate for open-sourcing?

**Comment by bjoernhaeuser** [comment]
I like very much!

File: pkg/provider/input.go [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+						},
+					},
+				},
+			},
+			"beats": {
+				Type:     schema.TypeSet,
+				Optional: true,
+
+				Elem: &schema.Resource{
+					// I didn't find any documentation about the GELF TCP input configuration.
```
Comment (bjoernhaeuser) [inline pkg/provider/input.go]:
```suggestion
					// I didn't find any documentation about the beats input configuration.
```

File: pkg/provider/input.go [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+						},
+					},
+				},
+			},
+			"gelf_http": {
+				Type:     schema.TypeSet,
+				Optional: true,
+
+				Elem: &schema.Resource{
+					// I didn't find any documentation about the GELF TCP input configuration.
```
Comment (bjoernhaeuser) [inline pkg/provider/input.go]:
```suggestion
					// I didn't find any documentation about the GELF Http input configuration.
```

**Pull request merged**